sympy/functions/special/bessel.py

```diff
@@ -685,14 +701,108 @@ class yn(SphericalBesselBase):
     def _rewrite(self):
         return self._eval_rewrite_as_bessely(self.order, self.argument)
 
+    def _eval_rewrite_as_besselj(self, nu, z):
+        return (-1)**(nu+1) * sqrt(pi/(2*z)) * besselj(-nu - S.Half, z)
+
     def _eval_rewrite_as_bessely(self, nu, z):
-        return sqrt(pi/(2*z)) * bessely(nu + S('1/2'), z)
+        return sqrt(pi/(2*z)) * bessely(nu + S.Half, z)
+
+    def _expand(self, **hints):
+        return _yn(self.order, self.argument)
+
+
+class SphericalHankelBase(SphericalBesselBase):
+
+    def _rewrite(self):
+        return self._eval_rewrite_as_besselj(self.order, self.argument)
+
+    def _eval_rewrite_as_besselj(self, nu, z):
+        # jn +- I*yn
+        # jn as beeselj: sqrt(pi/(2*z)) * besselj(nu + S.Half, z)
+        # yn as besselj: (-1)**(nu+1) * sqrt(pi/(2*z)) * besselj(-nu - S.Half, z)
+        hk = self._hankel_kind_sign
+        return sqrt(pi/(2*z))*(besselj(nu + S.Half, z) +
+                               hk*I*(-1)**(nu+1)*besselj(-nu - S.Half, z))
+
+    def _eval_rewrite_as_bessely(self, nu, z):
+        # jn +- I*yn
+        # jn as bessely: (-1)**nu * sqrt(pi/(2*z)) * bessely(-nu - S.Half, z)
+        # yn as bessely: sqrt(pi/(2*z)) * bessely(nu + S.Half, z)
+        hk = self._hankel_kind_sign
+        return sqrt(pi/(2*z))*((-1)**nu*bessely(-nu - S.Half, z) +
+                               hk*I*bessely(nu + S.Half, z))
 
     def _expand(self, **hints):
         n = self.order
         z = self.argument
-        return (-1)**(n + 1) * \
-               (fn(-n - 1, z) * sin(z) + (-1)**(-n) * fn(n, z) * cos(z))
+        hk = self._hankel_kind_sign
+
+        # fully expanded version
+        # return ((fn(n, z) * sin(z) +
+        #          (-1)**(n + 1) * fn(-n - 1, z) * cos(z)) +  # jn
+        #         (hk * I * (-1)**(n + 1) *
+        #          (fn(-n - 1, z) * hk * I * sin(z) +
+        #           (-1)**(-n) * fn(n, z) * I * cos(z)))  # +-I*yn
+        #         )
+
+        # call expand ?
+        return (_jn(n, z) + hk*I*_yn(n, z)).expand()
+
+
+class hn1(SphericalHankelBase):
+    r"""
+    Spherical Hankel function of the first kind.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, hn1, hankel1, expand_func
+    >>> z = Symbol("z")
+    >>> nu = Symbol("nu", integer=True)
+    >>> print(expand_func(hn1(0, z)))
+    sin(z)/z - I*cos(z)/z
+    >>> print(expand_func(hn1(1, z)))
+    -I*sin(z)/z - cos(z)/z + sin(z)/z**2 - I*cos(z)/z**2
+    >>> hn1(nu, z).rewrite(hankel1)
+    sqrt(2)*sqrt(pi)*sqrt(1/z)*hankel1(nu, z)/2
+
+    For integral orders :math:`n`, :math:`h_n^(1)` is calculated using the formula:
+
+    .. math:: h_n^(1)(z) = j_{n} + i (-1)^{n+1} j_{-n-1}(z)
```

Comment:
The mpmath docs say the definition is `J_n(x) + i Y_n(x)`, which seems more elegant..